gcovr/tests/test_gcov_parser.py

```diff
@@ -276,3 +280,39 @@ def test_pathologic_codeline(capsys):
         'IndexError',
         '(ERROR) Exiting',
         'run gcovr with --gcov-ignore-parse-errors')
+
+
+def check_and_raise(number, total, exc_raised, queue_full):
+    queue_full.wait()
+    if number == 0:
+        raise Exception("Number == 0")
+    exc_raised.wait()
+    total[0] += 1
+
+
+@pytest.mark.parametrize('threads', [1, 2, 4, 8])
+def test_pathologic_threads(capsys, threads):
+    mutable = [0]
+    queue_full = Event()
+    exc_raised = Event()
+    with pytest.raises(Exception) as excinfo:
+        with Workers(threads, lambda: {'total': mutable, 'exc_raised': exc_raised, 'queue_full': queue_full}) as pool:
+            for extra in range(0, 10000):
+                pool.add(check_and_raise, extra)
+
+            # Queue is filled
+            queue_full.set()
+
+            # Wait until the exception has been completed
+            while not pool.exceptions:
+                time.sleep(0)
+
+            # Queue should be drained and exception raised
+            exc_raised.set()
+            pool.wait()
+
+    # Outer level catches correct exception
+    assert excinfo.value.args[0] == "Number == 0"
+
+    # Fewer than 1 job per thread is executed
+    assert mutable[0] < threads
```

Comment:
I think it would be possible that exactly one job per thread is dequeued. Should the check be changed to “At most 1 job per thread”: `assert mutable[0] <= threads`?

Reply:
That is true - but one of those jobs is to throw an exception, so it doesn't increment the counter. ...which is not even slightly explained by my comment! Again, I could change the comment prior to the merge to say "increment" instead of "job"?